src/http/Response.js

```diff
@@ -71,4 +71,19 @@ export default class Response extends Message {
     prefix(messageWriter) {
         return messageWriter.responsePrefix(this);
     }
+
+    // returns the 'ID' field value of the response copied to from the request (if any)
+    extractMatchingId(request) {
+        const idFieldName = request._daftFieldName("ID");
+        if (this.header.has(idFieldName))
+            return this.header.value(idFieldName);
+        return null;
+    }
+
+    // copy the 'ID' field from the received request
```

Comment:
```suggestion
    // Copy the Daft request ID field (if any) from the given request.
    // The requestId() method can be used to extract the copied ID.
```

Adjust as needed depending on what happens to extractMatchingId().